services/web/server/src/simcore_service_webserver/projects/projects_db.py

```diff
@@ -724,6 +739,10 @@ def _update_workbench(
                     .returning(literal_column("*"))
                 )
                 project: RowProxy = await result.fetchone()
+
+                for frontend_node_update_task in frontend_nodes_update_tasks:
```

Comment:
This is the main cause of the circular import discussed in https://github.com/ITISFoundation/osparc-simcore/pull/3058/files#r880401810

IMO the db-API layer for projects should know nothing about front-end nodes etc . The idea here is create a repository pattern (SEE e.g. https://www.cosmicpython.com/book/chapter_02_repository.html) whose responsibility is accessing the database and hiding all sql-alchemy specialized logic.

The operations you are doing here IMO belong outside ``projects_db``, where the concept of e.g. front-end-node is well defined but not here, where the only things that are defined are the columns of the table etc

Reply:
I managed to move it outside in the end. You are correct, those functions should have not been inside the `projects_db` since it has a different objective.